### PR TITLE
Allow client to provide the OpenAIClient and APIM example

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

Please help reviewers and future users, providing the following information:
  1. Why is this change required? Demonstrates how to use SK together with Azure API Management
  2. What problem does it solve? Allows a custom OpenAIClient to be created by client code and thereby have full access to `OpenAIClientOptions`
  3. What scenario does it contribute to? Support for customer who want to protect their Azure OpenAI API keys 
  4. If it fixes an open issue, please link to the issue here.

### Description
Allowing the OpenAIClient to be created by the client gives them full access to the `OpenAIClientOptions`.
Consider the case where a developer is using APIM to protect their API keys and starts to see a problem. 
If you run the associated sample with a bad subscription key the trace in will include the following:

```
      Headers:
      ErrorSource: authorization
      ErrorReason: SubscriptionKeyInvalid
      ErrorMessage: Access denied due to invalid subscription key. Make sure to provide a valid key for an active subscription.
      ErrorSection: backend
      ErrorStatusCode: 401
```

Without the `Diagnostics` seeting in `OpenAIClientOptions` the trace in will include the following:

```
      Headers:
      ErrorSource: REDACTED
      ErrorReason: REDACTED
      ErrorMessage: REDACTED
      ErrorSection: REDACTED
      ErrorStatusCode: REDACTED
```

### Contribution Checklist
- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: samples/dotnet/kernel-syntax-examples/Program.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@ public static class Program
     // ReSharper disable once InconsistentNaming
     public static async Task Main()
     {
+        /*
```
Comment (lemillermicrosoft) [inline samples/dotnet/kernel-syntax-examples/Program.cs:11]:
```suggestion
```

File: samples/dotnet/kernel-syntax-examples/Program.cs [hunk 2]
```diff
@@ -148,5 +149,9 @@ public static async Task Main()
 
         await Example47_Redis.RunAsync();
         Console.WriteLine("== DONE ==");
+        */
```
Comment (lemillermicrosoft) [inline samples/dotnet/kernel-syntax-examples/Program.cs:152]:
```suggestion
```

**Review by lemillermicrosoft — CHANGES_REQUESTED** [review]
LGTM. One change needed. This should help enable some better unit test coverage I think, too!

File: .github/workflows/java-format.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+name: Java format automation
```
Comment (lemillermicrosoft) [inline .github/workflows/java-format.yml:1]: I'm fine with these changes, but are they intentional with this pr?

File: dotnet/src/Connectors/Connectors.AI.OpenAI/TextEmbedding/AzureTextEmbeddingGeneration.cs [hunk 4]
```diff
@@ -12,12 +13,12 @@
 namespace Microsoft.SemanticKernel.Connectors.AI.OpenAI.TextEmbedding;
 
 /// <summary>
-/// Azure OpenAI text embedding service.
```
Comment (lemillermicrosoft) [inline dotnet/src/Connectors/Connectors.AI.OpenAI/TextEmbedding/AzureTextEmbeddingGeneration.cs:15]: I think service was the right term we're going with as `ITextEmbeddingGeneration` implements `IAIService`

Comment (lemillermicrosoft) [reply dotnet/src/Connectors/Connectors.AI.OpenAI/TextEmbedding/AzureTextEmbeddingGeneration.cs:15]: I see though we were not consistent in this file already. /shrug

File: python/.conf/.pre-commit-config.yaml [hunk 5]
```diff
@@ -15,15 +15,8 @@ repos:
     hooks:
```
Comment (lemillermicrosoft) [inline python/.conf/.pre-commit-config.yaml:15]: These changes intentional?

Comment (markwallace-microsoft) [reply python/.conf/.pre-commit-config.yaml:15]: Created a new PR: https://github.com/microsoft/semantic-kernel/pull/1520

**Pull request closed without merging**